addons/mass_mailing/static/src/js/mass_mailing_wysiwyg.js

```diff
@@ -70,6 +91,34 @@ export class MassMailingWysiwyg extends Wysiwyg {
         this.mainToolbarEl.classList.toggle('d-none', !fold);
     }
 
+    /**
+     * @override
+     */
+    onIframeUpdated() {
+        super.onIframeUpdated();
+        const iframe = this.$iframe[0];
+        if (!iframe || !iframe.isConnected) {
+            return;
+        }
+        const body = iframe.contentWindow.document.body;
+        if (iframe.classList.contains("has_snippets_sidebar")) {
+            body.classList.add("o_mass_mailing_iframe_body_with_snippets_sidebar");
+        }
+    }
+
+    /**
+     * @override
+     */
+    onSnippetsFoldChange(folded) {
+        super.onSnippetsFoldChange(folded);
+        const iframe = this.$iframe?.[0];
+        if (!iframe || !iframe.isConnected) {
+            return;
+        }
+        const body = iframe.contentWindow.document.body;
+        body.classList.toggle("o_mass_mailing_iframe_body_with_snippets_sidebar", !folded);
```

Comment:
A bit sad that we're repeating this code.

Could we create some kind of state variable, like "isSnippetsVisible", and condition some code based on the fact that it changes?
Maybe it's overkill, just asking.